spec/unit/resource_spec.rb

```diff
@@ -315,8 +323,8 @@
 
       run_context.resource_collection << Chef::Resource::ZenMaster.new("bean")
```

Comment:
try testing this with the call to `.new("bean", run_context)` and see if the nil error goes away

Reply:
Yes error goes away if we add run_context but then if we keep below section as it is, it will be logically incorrect?
`zrb.subscribes :reload, zr
      expect(zr.delayed_notifications.detect { |e| e.resource.name == resource.name && e.action == :reload }).not_to be_nil`
  As in, it would mean we want `zrb` to be the resource subscribing to `zr`, the resource `resource` is not used here so, the `.detect{}` should check for condition `e.resource.name == zr.name`

Reply:
i think we probably just want to add the `run_context` to all the calls to build resources and probably leave the rest of the code as it stands.

Reply:
Updated. Also branch is rebased with main